lib/api/src/grpc/transport_channel_pool.rs

```diff
@@ -35,35 +89,72 @@ impl TransportChannelPool {
 
     /// Initialize a pool for the URI and return a clone of the first channel.
     /// Does not fail if the pool already exist.
-    async fn init_pool_for_uri(&self, uri: Uri) -> Result<Channel, Error> {
+    async fn init_pool_for_uri(&self, uri: Uri) -> Result<Channel, TonicError> {
         let mut guard = self.uri_to_pool.write().await;
         match guard.get(&uri) {
             None => {
-                let mut channels = Vec::with_capacity(self.pool_size);
-                for _ in 0..self.pool_size {
-                    let channel = Self::make_channel(self.grpc_timeout, uri.clone()).await?;
-                    channels.push(channel);
-                }
-                let result = channels[0].clone();
-                guard.insert(uri.clone(), channels);
-                Ok(result)
+                let channels =
+                    ChannelsToUri::init(uri.clone(), self.pool_size, self.grpc_timeout).await?;
+                let channel = channels.choose();
+                guard.insert(uri, channels);
+                Ok(channel)
             }
-            Some(channels) => Ok(channels[0].clone()),
+            Some(channels) => Ok(channels.choose()),
         }
     }
 
-    pub async fn get_pooled_channel(&self, uri: &Uri) -> Option<Channel> {
+    async fn drop_pool(&self, uri: &Uri) {
+        let mut guard = self.uri_to_pool.write().await;
+        guard.remove(uri);
+    }
+
+    async fn get_pooled_channel(&self, uri: &Uri) -> Option<Channel> {
         let guard = self.uri_to_pool.read().await;
-        guard
-            .get(uri)
-            .and_then(|channels| channels.choose(&mut rand::thread_rng()))
-            .cloned()
+        guard.get(uri).map(|channels| channels.choose())
     }
 
-    pub async fn get_or_create_pooled_channel(&self, uri: &Uri) -> Result<Channel, Error> {
+    async fn get_or_create_pooled_channel(&self, uri: &Uri) -> Result<Channel, TonicError> {
         match self.get_pooled_channel(uri).await {
             None => self.init_pool_for_uri(uri.clone()).await,
             Some(channel) => Ok(channel),
         }
     }
+
+    async fn get_last_created(&self, uri: &Uri) -> Option<Instant> {
```

Comment:
Again nitpicking on naming sorry.

I'd prefer something like `get_created_at` as it retrieves an `Instant` and not the last channel.